src/lightning/fabric/strategies/fsdp.py

```diff
@@ -491,9 +491,18 @@ def save_checkpoint(
                         target_dict = metadata
                     _apply_filter(key, filter or {}, converted, target_dict)
 
-            # FSDP's FileSystemWriter streams the tensors to disk to minimize memory peaks
-            writer = FileSystemWriter(path=path, single_file_per_rank=True)
-            save_state_dict(converted_state, writer)
+            if _TORCH_GREATER_EQUAL_2_3:
+                from torch.distributed.checkpoint import save
+                save(converted_state, checkpoint_id=path)  # type: ignore[call-arg]
+            else:
+                from torch.distributed.checkpoint import FileSystemWriter, save
+                if _TORCH_GREATER_EQUAL_2_2:
+                    from torch.distributed.checkpoint import save
+                else:
+                    from torch.distributed.checkpoint import save_state_dict as save
+                # FSDP's FileSystemWriter streams the tensors to disk to minimize memory peaks
+                writer = FileSystemWriter(path=path, single_file_per_rank=True)
+                save(converted_state, writer)
```

Comment:
would it make sense to refactor this into a function so that the already huge abomination of this method stays somewhat readable 😅

Reply:
Done